### PR TITLE
image color

### PR DESCRIPTION
**Pull request by Omid07**

### CODE REVIEW

File: app/src/main/java/com/example/omid/ps01/MainFragment.java [hunk 1]
```diff
@@ -17,17 +17,20 @@
...
+    private Button mButtonColor;
 
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
         View view = inflater.inflate(R.layout.fragment_main, container, false);
         mImageView = (ImageView) view.findViewById(R.id.image_edit);
         mButtonCrop = (Button) getActivity().findViewById(R.id.button_crop);
-        mButtonCrop.setVisibility(View.VISIBLE);
+        mButtonLight = (Button) getActivity().findViewById(R.id.button_light);
```
Comment (nvmanh) [inline app/src/main/java/com/example/omid/ps01/MainFragment.java]: why all logic code in here, how's about onViewCreated, onActivityCreated, with findViewByID to initialize view object break to a separate method pls

File: app/src/main/java/com/example/omid/ps01/ColorFragment.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        mButtonBlue.setOnClickListener(this);
+        mButtonGreen.setOnClickListener(this);
+        return view;
+    }
+
+    @Override
+    public void onActivityCreated(Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+        mButtonCrop = (Button) getActivity().findViewById(R.id.button_crop);
+        mButtonLight = (Button) getActivity().findViewById(R.id.button_light);
```
Comment (nvmanh) [inline app/src/main/java/com/example/omid/ps01/ColorFragment.java]: i mention you should create a method for findview functions not put them here, here is for your logic code

File: app/src/main/java/com/example/omid/ps01/MainFragment.java [hunk 3]
```diff
@@ -39,6 +35,18 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
     }
 
     @Override
+    public void onActivityCreated(Bundle savedInstanceState) {
+        super.onActivityCreated(savedInstanceState);
+        mButtonCrop = (Button) getActivity().findViewById(R.id.button_crop);
```
Comment (nvmanh) [inline app/src/main/java/com/example/omid/ps01/MainFragment.java]: same

File: app/src/main/java/com/example/omid/ps01/EditImageActivity.java [hunk 4]
```diff
@@ -78,4 +98,9 @@ public void getCropImage(Bitmap bitmap) {
     public void getLightImage(Bitmap bitmap) {
         mOriginalBitmap = bitmap;
     }
+
+    @Override
+    public void getColorImage(Bitmap bitmap) {
+        mOriginalBitmap = bitmap;
```
Comment (nvmanh) [inline app/src/main/java/com/example/omid/ps01/EditImageActivity.java]: have you ever recycle bitmap for releasing memory?

Comment (Omid07) [inline app/src/main/java/com/example/omid/ps01/EditImageActivity.java]: @nvmanh no i haven't done it before.. should i use Caching Bitmaps for this ?
https://developer.android.com/training/displaying-bitmaps/cache-bitmap.html

Comment (nvmanh) [inline app/src/main/java/com/example/omid/ps01/EditImageActivity.java]: https://developer.android.com/training/displaying-bitmaps/manage-memory.html
read it pls

File: app/src/main/java/com/example/omid/ps01/EditImageActivity.java [hunk 5]
```diff
@@ -70,12 +89,29 @@ public String getImagePath() {
...
+    }
+
+    @Override
+    public void getColorImage(Bitmap bitmap) {
+        recycling();
         mOriginalBitmap = bitmap;
     }
+
+    public void recycling()
```
Comment (nvmanh) [inline app/src/main/java/com/example/omid/ps01/EditImageActivity.java]: misunderstand, when destroy this fragment or activity, recycle bitmap if not use

**Comment by nvmanh** [comment]
OK!

**Pull request merged**